### PR TITLE
Extend block parameter methods to accept 'finalized' and 'safe' as block tags

### LINKED ISSUE #3947: Extend eth_getBlockByNumber to support finalized and safe as block tags
### Description
For The Merge the `eth_getBlockByNumber` must accept 2 new block tags: `finalized` and `safe`, as described by the spec:

>  `finalized`: The most recent crypto-economically secure block, cannot be re-orged outside of manual intervention driven by community coordination; 
> `safe`: The most recent block that is safe from re-orgs under honest majority and certain synchronicity assumptions;

### PR DESCRIPTION
**Pull request by fab-10**
Signed-off-by: Fabio Di Fabio <fabio.difabio@consensys.net>

## Fixed Issue(s)

fixes #3947 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Additions and Improvements
 - \[EXPERIMENTAL\] Add checkpoint sync `--sync-mode="X_CHECKPOINT"` [#3849](https://github.com/hyperledger/besu/pull/3849)
+- Support `latest` and `safe` as tags for the block parameter in RPC APIs [#3950](https://github.com/hyperledger/besu/pull/3950)
```
Comment (shemnon) [inline CHANGELOG.md:7]: s/latest/finalized/

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AbstractBlockParameterMethod.java [hunk 2]
```diff
@@ -57,21 +60,44 @@ protected Object latestResult(final JsonRpcRequestContext request) {
...
     final Optional<Long> blockNumber = blockParam.getNumber();
     if (blockNumber.isPresent()) {
-      result = resultByBlockNumber(request, blockNumber.get());
-    } else if (blockParam.isLatest()) {
-      result = latestResult(request);
-    } else {
-      // If block parameter is not numeric or latest, it is pending.
-      result = pendingResult(request);
+      return resultByBlockNumber(request, blockNumber.get());
+    }
```
Comment (shemnon) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AbstractBlockParameterMethod.java:89]: Style wise these should remain if/else chained.  They are not proper "guard blocks" and readability would be improved by retaining the if/elseif/elseif chain.  Bringing in the last return as an else would further improve the readability.

**Review by shemnon — APPROVED** [review]
LGTM presuming a  change in the changelog and restoring the 'if/else if' chaining

File: CHANGELOG.md (hunk 1 above)
Comment (fab-10) [reply CHANGELOG.md]: done

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AbstractBlockParameterMethod.java (hunk 2 above)
Comment (fab-10) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AbstractBlockParameterMethod.java]: done

**Review by shemnon — APPROVED** [review]

**Comment by fab-10** [comment]
> ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AbstractBlockParameterMethod.java

what do you mean?

**Pull request merged**